File: cubord-backend/src/main/java/org/cubord/cubordbackend/domain/Location.java
package org.cubord.cubordbackend.domain;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "locations")
@Getter
@Setter
@ToString(exclude = "household")
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(of = "id")
public class Location {

    @Id
    private UUID id;

    private String name;

    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "household_id")
    private Household household;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: cubord-app/src/types/Location.ts
export interface LocationRequest {
    /** @maxLength 255 */
    name: string;
    /** @maxLength 500 */
    description?: string | null;
    householdId: string;
}

export interface LocationUpdateRequest {
    /** @maxLength 255 */
    name: string;
    /** @maxLength 500 */
    description?: string | null;
}

export interface LocationResponse {
    id: string;
    name: string;
    description: string | null;
    householdId: string;
    householdName: string | null;
    createdAt: string;
    updatedAt: string;
}